.vib/nats/ginkgo/nats_test.go

```diff
@@ -78,26 +83,28 @@ var _ = Describe("NATS", Ordered, func() {
 				return c.BatchV1().Jobs(namespace).Get(ctx, putJobName, getOpts)
 			}, timeout, PollingInterval).Should(WithTransform(getSucceededJobs, Equal(int32(1))))
 
-			// Give the application some time to sync the data
-			time.Sleep(10*time.Second)
-
-			By("scaling down to 0 replicas")
-			ss, err = utils.StsScale(ctx, c, ss, 0)
+			By("deleting the job once it has succeeded")
+			err = c.BatchV1().Jobs(namespace).Delete(ctx, putJobName, metav1.DeleteOptions{})
 			Expect(err).NotTo(HaveOccurred())
 
-			Eventually(func() (*appsv1.StatefulSet, error) {
-				return c.AppsV1().StatefulSets(namespace).Get(ctx, stsName, getOpts)
-			}, timeout, PollingInterval).Should(WithTransform(getAvailableReplicas, BeZero()))
+			// Give the application some time to sync the data
+			time.Sleep(10 * time.Second)
 
-			By("scaling up to the original replicas")
-			ss, err = utils.StsScale(ctx, c, ss, origReplicas)
+			By("rollout restart the statefulset")
+			_, err = utils.StsRolloutRestart(ctx, c, ss)
 			Expect(err).NotTo(HaveOccurred())
 
+			for i := 0; i < int(origReplicas); i++ {
```

Comment:
The restart command goes from N-1 to 0. If the startup of the nodes takes long, you may reach the timeout because node 0 has not been restarted yet. In other charts we do the loop backwards

Reply:
done at https://github.com/bitnami/charts/pull/29462/commits/2712215638e2cf11d9eca92601af791f5df80eb5